Design/reflectometry-idf.md

```diff
@@ -0,0 +1,62 @@
+# Reflectometry IDF
+
+## Motivation
+
+The reflectometry instrument scientists do not currently use an IDF and are heavily reliant on their own scripts.
+If the IDF could properly describe their setup then it would solve existing issues and improve their workflow.
+
+Potential benefits include, but are not limited to:
+  - Correct geometric description of detector components and calculation of angles to be fed into [algorithms](https://github.com/mantidproject/mantid/issues/26971)
+  - Complex beam paths to allow for accurate description of mirror action, with a view to accurately describe the effect of gravity
+  - Saving time by having multiple modes of configuration described within one file
+  - Reduce known sources of error due to problems with Q error bars, incorrect angles, gravity corrections, etc.
+  - Facilitate new equipment over multiple instruments/facilities, such as 2D detectors
+
+## Solution
+
+This task should be thought of in terms of four main sub-tasks:
+
+  ### 1. POLREF IDF support for relative pixel map
+  
+  Andrew currently has a scan of the POLREF detector which he uses in a script to move each detector pixel into place every time 
+  they run a reduction. This appears to be something which can be supported in an IDF by obtaining the relative pixel positions 
+  from the scan and hard-coding them. The actual specular pixel position could then be obtained from the logs and used to do this 
+  rotation into place from within the IDF.
+  
+  ### 2. Support multiple L-distances through a complex beam-path
+  
+  Current workarounds include the following:
+    - Separate 'neutronic' and 'physical instrument' IDFs
+	- Dynamic modification of positions via `MoveInstrumentComponent`
+	- Storage of the L1 & L2 distances in a separate table, which `ConvertUnits` can be made to 'understand'
+	
+  The ideal solution to this task would be to create a new `beam-path` tag for description within the IDF. This can then can be used 
+  with Owen’s [Instrument 2.0 prototype](https://github.com/DMSC-Instrument-Data/instrument-prototype) which specifically deals with 
+  the in-memory side of this task. He has mentioned that ray tracing will not be possible, and even with the prototype discussed above 
+  only one flight path per component has been considered, so additional thought will need to be given to multiple flight paths as this 
```

Comment:
Yes. That is the limitation. From your description of their procedure above, the complex beam path approach might work for them. Given that they are using L2,L1 tables or neutronic/physical instrument IDFs I think they are already imposing the restriction on themselves of one flight path per component.

Reply:
I don't they are using these workarounds. I added them as background after talking to you and Nick about existing solutions that other instruments use, but maybe that is confusing and needs to be made explicit.